podvm/Makefile

```diff
@@ -29,11 +29,13 @@ ifeq ($(PODVM_DISTRO),ubuntu)
 	-var cloud_image_url=${UBUNTU_IMAGE_URL} \
 	-var cloud_image_checksum=${UBUNTU_IMAGE_CHECKSUM})
 ifneq ($(HOST_ARCH),$(ARCH))
-	$(eval OPTS += -var qemu_binary=qemu-system-${ARCH} \
-	-var machine_type=${QEMU_MACHINE_TYPE_${ARCH}} \
+	$(eval OPTS += -var machine_type=${QEMU_MACHINE_TYPE_${ARCH}} \
 	-var boot_wait=300s)
+ifndef QEMU_BINARY
+	$(eval OPTS += -var qemu_binary=qemu-system-${ARCH})
```

Comment:
I know this WIP but does this work ? I recall the right way is to create an user variable and then do something like qemu_binary = S{var.qemu_image_binary} in the packer script.

Reply:
It does work (it  was here before), i think the  reason is because in the packer files we cannot applying this according to some logic (identify arch in this case)

Reply:
Looking here - https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/podvm/qcow2/ubuntu/variables.pkr.hcl, qemu_binary is already defined as a user variable and that's why this works. Although not related to this MR and is mostly a nit: please consider renaming the user variable qemu_binary to distinguish it from the packer variable qemu_binary.

Reply:
TBH i'm not sure why we set it as it's the [default of packer anyhow](https://developer.hashicorp.com/packer/plugins/builders/qemu#qemu_binary)

Reply:
Yeah, that's what I was referring to in the first comment. The "-var ..." syntax is only for user defined variables. You cannot set a predefined variable that way - that has to go in the main script. That's why the main script has `qemu_binary=$(var.qemu_binary}`

Reply:
It's easier to read the template if I know which of them are user defined variables and which of them predefined. It's actually confusing to someone new to packer trying to do something similar.

Reply:
Maybe I'm confused, what would be the alternative? Is there a better way to update that value?

I don't see the difference between `var.qemu_binary` and `var.qemu_image_binary` to tell me if it is user defined or not.

Reply:
Ofcourse there is :) qemu_image_binary is not mentioned in https://developer.hashicorp.com/packer/plugins/builders/qemu. So, that's a cue for someone to go look where it came from. Anyway, as I  mentioned in my previous comment, this was a nit but generally, I believe mixing user defined variables and predefined variables makes the code confusing to read.